### DISCUSSION TITLE
Development Environments

### DISCUSSION THREAD

**Discussion by 0x4007** [pavlovcik/uad-ui#673] [General]
### Discussed in https://github.com/ubiquity/ubiquity-dollar/discussions/652

<div type='discussions-op-text'>

<sup>Originally posted by **rndquu** May 24, 2023</sup>
Let's use different environments for https://uad.ubq.fi/

I mean:
1. setup https://uad.ubq.fi/ to track `ubiquity/ubiquity-dollar#main` branch for contracts and `ubiquity/ubq.fi#main` branch for UI
2. setup https://dev.uad.ubq.fi/ to track `ubiquity/ubiquity-dollar#development` branch for contracts and `ubiquity/ubq.fi#development` branch for UI
3. setup https://stage.uad.ubq.fi/ to track `ubiquity/ubiquity-dollar#release-X.X.X` branch for contracts and `ubiquity/ubq.fi#X.X.X` branch for UI. Here the `release-X.X.X` branch is the branch that is QAed right now and is about to merged to the `main` branch and deployed to production

This way we could setup different deployment scripts for ubiquity contracts depending on the environment

What do you think @ubiquity/development ?</div>

**Comment by 0x4007** [comment]
Are you able to walk me through how to do these on Cloudflare? I believe that `1.` and `2.` are simple CNAME DNS entries, but I believe that `3.` will require some type of custom Cloudflare Worker? 

I suggest you use any domain name you own, and try setting up your idea there. Then I can copy your settings and reflect them on https://ubq.fi

**Comment by 0x4007** [comment]
@rndquu please update the time estimate if you think my estimate is incorrect. I just made a guess based on this specification and [your comment](https://github.com/ubiquity/ubiquity-dollar/issues/322#issuecomment-1578086208).

**Reply by rndquu** [reply]
I converted this issue back to the discussion. I will break it into smaller tasks.
